storehaus-core/src/main/scala/com/twitter/storehaus/FanoutStore.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2014 Twitter inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.twitter.storehaus
+
+import com.twitter.algebird.{Semigroup, Monoid}
+import com.twitter.util.Future
+
+/**
+ * A store that fans out Key K to a Set of Keys, queries the appropriate underlying stores and sum
+ * the values using the Monoid V
+ *
+ * @author Mansur Ashraf
+ */
+class FanoutStore[-K, +V: Semigroup, S <: ReadableStore[K, V]](fanout: K => Set[K], stores: Set[(K => Boolean, S)])
```

Comment:
This is a cool idea, but question: why not make it:
```scala
class FanoutStore[-K, K1, +V: Semigroup, S <: ReadableStore[K1, V]](fanout: K => Iterable[K1], stores: K1 => Iterable[S])
```
the `Set[(LK => Boolean, S)]` seems confusing to me. It seems you just want a lookup function, right?